test/test_testing.py

```diff
@@ -123,6 +123,95 @@ def test_update_relation_exposes_new_data(self):
         self.assertEqual(viewer.changes, [{'initial': 'data'},
                                           {'initial': 'data', 'new': 'value'}])
 
+    def test_update_relation_no_local_unit_change_event(self):
+        # language=YAML
+        harness = Harness(CharmBase, meta='''
+            name: my-charm
+            requires:
+              db:
+                interface: pgsql
+            ''')
+        harness.begin()
+        helper = DBRelationChangedHelper(harness.charm, "helper")
+        rel_id = harness.add_relation('db', 'postgresql')
+        harness.update_relation_data(rel_id, 'my-charm/0', {'new': 'value'})
+        rel = harness.charm.model.get_relation('db')
+        # Our unit data bag got updated.
+        self.assertTrue(rel.data[harness.charm.model.unit]['new'], 'value')
+        # But there were no changed events registered by our unit.
+        self.assertTrue(len(helper.changes) == 0)
```

Comment:
I prefer
self.assertEqual([], helper.changes)

because then you can see what changes *did* happen in the traceback, rather than just 'the count was not 0' which is very hard to debug what did happen and why it was wrong.

Reply:
Good point, I'll change it.

Reply:
https://github.com/canonical/operator/pull/212/commits/516c13393c0c4d4222aa2d85f520d0e37ecb0860